pkgs/top-level/all-packages.nix

```diff
@@ -13442,7 +13442,7 @@ in
   boost174 = callPackage ../development/libraries/boost/1.74.nix { };
   boost175 = callPackage ../development/libraries/boost/1.75.nix { };
   boost17x = boost175;
-  boost = boost16x;
+  boost = boost17x;
```

Comment:
I am not sure about this change. It is probably going to break a lot and I can't really test it right now because my usual test packages are not building on staging.

Reply:
I've had a hard time testing this due to the sheer number of packages that need to be re-built. If someone with the storage space and will to do a comprehensive nixpkgs-review test on it (or some other mechanism of seeing what packages are broken), I can do a regex replace boost -> boost16x for anything that wont build with the newer version.

Reply:
I have a quite beefy remote build farm available but I can't build 5000+ packages. I don't know how we should continue with this PR.

Reply:
Targeting at a merge base of staging and master makes more sense, there binary cache should be available for testing.